static/js/unread_ops.js

```diff
@@ -73,6 +74,53 @@ export function process_read_messages_event(message_ids) {
     unread_ui.update_unread_counts();
 }
 
+export function process_unread_messages_event({message_ids, message_details}) {
+    // This is the reverse of  process_unread_messages_event.
+    message_ids = unread.get_read_message_ids(message_ids);
+    if (message_ids.length === 0) {
+        return;
+    }
+
+    for (const message_id of message_ids) {
+        const message = message_store.get(message_id);
+
+        if (message) {
+            message.unread = true;
+        }
+
+        const message_info = message_details[message_id];
+
+        let user_ids_string;
+
+        if (message_info.type === "private") {
+            user_ids_string = message_info.sender_id.toString();
+        } else if (message_info.type === "huddle") {
+            user_ids_string = people.pm_lookup_key(message_info.sender_id_str);
+        }
+
+        unread.process_unread_message({
+            id: message_id,
+            mentioned: message_info.mentioned,
+            stream_id: message_info.stream_id,
+            topic: message_info.topic,
+            type: message_info.type === "stream" ? "stream" : "private",
+            unread: true,
+            user_ids_string,
+        });
+
+        if (message_info.type === "stream") {
+            recent_topics_ui.update_topic_unread_count({
+                stream_id: message_info.stream_id,
+                topic: message_info.topic,
+            });
+        }
+    }
+
+    message_lists.current.rerender_view();
```

Comment:
Doing a full rerender is a big hammer. That's potentially OK for a new, infrequent operation, but we should document that this is an expediency choice in a comment, and that we'll want to add a function like the existing `message_list_view` function `show_message_as_read`. 

I also suspect we need to do a rerender of `message_lists.home` as well if different, so that hitting `Esc` shows the correct state of the "All messages" view if you have that as your default view. We probably (should) have a helper for doing such a "rerender all messages lists"; yeah `message_list_update.rerender_messages_view` will do it (not currently exported).

Reply:
This is the fix I am testing.  (the comment may get tweaked when I push, but it's the gist)
~~~
135     /*
136         We use a big-hammer approach now to updating the message view.
137         This is relatively harmless, since the only time we are called is
138         when the user herself marks her message as unread.  But we
139         do eventually want to be more surgical here, especially once we
140         have a final scheme for how best to structure the HTML within
141         the message to indicate read-vs.-unread.  Currently we use a
142         green border, but that may change.
143     */
144     message_live_update.rerender_messages_view();

Reply:
Yeah, sgtm.